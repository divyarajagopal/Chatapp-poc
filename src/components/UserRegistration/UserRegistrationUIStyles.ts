import { StyleSheet } from 'react-native';
export const UserRegistrationStyles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  inputContainer: {
    width: '80%'
  },
  loader: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'center'
  }
});
