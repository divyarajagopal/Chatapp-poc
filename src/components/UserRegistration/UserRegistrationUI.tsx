import * as React from 'react';
import { Component } from 'react';
import { View, Button, ActivityIndicator, AsyncStorage } from 'react-native';
import { UserRegistrationStyles as styles } from './UserRegistrationUIStyles';
import { IUserRegistrationRender } from '../../__common__/containers/UserRegistration/UserRegistrationRender';
import { IChatSession } from '../../models/ChatSession';
import { AsyncStorageKeys } from '../../constants/AsyncStorageKeys';
import { ConnectedUserRegistration } from '../../__common__/containers/UserRegistration/UserRegistration';
import TextInputUI from '../controls/TextInputUI';
import TouchableOpacityUI from '../controls/TouchableOpacityUI';

class UserRegistrationUI extends Component<any, any> {
  constructor(props: any) {
    super(props);
  }

  public onStartChatPressed = async (renderObject: IUserRegistrationRender) => {
    renderObject.onStartChatPressed();

    const chatSessionObj: IChatSession = {
      username: renderObject.state.userName,
      email: renderObject.state.email
    };
    await AsyncStorage.setItem(
      AsyncStorageKeys.CHAT_SESSION,
      JSON.stringify(chatSessionObj)
    );
  };

  public render = () => {
    const renderLoader = (isLoaded?: boolean) => {
      let loader = <View />;
      if (!isLoaded) {
        loader = (
          <View style={styles.loader}>
            <ActivityIndicator size="large" />
          </View>
        );
      }
      return loader;
    };

    return (
      <View style={{ flex: 1 }}>
        <ConnectedUserRegistration
          navigation={this.props.navigation}
          render={(renderObj: IUserRegistrationRender) => (
            <View style={styles.container}>
              {renderLoader(renderObj.props && renderObj.props.isLoaded)}
              <View style={styles.inputContainer}>
                <TextInputUI
                  error={
                    renderObj.state.userName && !renderObj.state.isValidUserName
                  }
                  value={renderObj.state.userName}
                  placeholder="Username"
                  onChangeText={renderObj.onUsernameChangedHandler}
                />
                <TextInputUI
                  error={renderObj.state.email && !renderObj.state.isValidEmail}
                  value={renderObj.state.email}
                  placeholder="Email"
                  onChangeText={renderObj.onEmailChangedHandler}
                />
                <TouchableOpacityUI>
                  <Button
                    disabled={!renderObj.state.enableStartChatBtn}
                    title="Start Chat"
                    onPress={async () =>
                      await this.onStartChatPressed(renderObj)
                    }
                  />
                </TouchableOpacityUI>
              </View>
            </View>
          )}
        />
      </View>
    );
  };
}

export default UserRegistrationUI;
