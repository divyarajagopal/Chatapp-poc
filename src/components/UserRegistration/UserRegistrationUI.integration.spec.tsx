/**
 * @jest-environment jsdom
 */
import * as React from 'react';
import { IApplicationStoreState } from '../../__common__';
import configureStore from 'redux-mock-store';
import { mount, configure, ReactWrapper } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import UserRegistrationUI from './UserRegistrationUI';
import { Provider } from 'react-redux';
import { UserRegistration } from '../../__common__/containers/UserRegistration/UserRegistration';
import { View } from 'react-native';
import '';

describe('UserRegistrationUI Component integration with child component tests', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore();
  let store: any;
  let wrapper: ReactWrapper;
  let userRegistration: UserRegistration;
  let userRegistrationWrapper: any;
  const originalConsoleError = console.error;

  afterAll(() => {
    console.error = originalConsoleError;
  });

  beforeAll(() => {
    console.error = (message: any) => {
      if (message.startsWith('Warning:')) {
        return;
      }

      originalConsoleError(message);
    };
  });

  beforeEach(() => {
    const applicationState: IApplicationStoreState = {
      welcomeStoreState: {
        data: [],
        errorMessage: undefined,
        isLoaded: false
      }
    };
    store = mockStore(applicationState);
    wrapper = mount(
      <Provider store={store}>
        <UserRegistrationUI />
      </Provider>
    );

    userRegistrationWrapper = wrapper.find(UserRegistration);

    userRegistration = userRegistrationWrapper.instance() as UserRegistration;
  });

  test('should render the component as expected with store context', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should validate UserName', () => {
    userRegistration.onUsernameChangedHandler('test');
    expect(userRegistrationWrapper.state('isValidUserName')).toBeTruthy();

    userRegistration.onUsernameChangedHandler(
      '1234567890123456789012345678901'
    );
    expect(userRegistrationWrapper.state('isValidUserName')).toBeFalsy();
  });

  test('should validate Email', () => {
    userRegistration.onEmailChangedHandler('test@test.com');
    expect(userRegistrationWrapper.state('isValidEmail')).toBeTruthy();

    userRegistration.onEmailChangedHandler('invalid email');
    expect(userRegistrationWrapper.state('isValidEmail')).toBeFalsy();
  });

  test('should set the data on pressing start chat button', () => {
    const expectedUserName = 'test';
    expect(userRegistration.props.isLoaded).toBeFalsy();

    userRegistration.componentDidMount();

    userRegistration.onUsernameChangedHandler(expectedUserName);
    userRegistration.onEmailChangedHandler('test@test.com');
    expect(userRegistrationWrapper.state('enableStartChatBtn')).toBeTruthy();

    userRegistration.onStartChatPressed();
    expect(userRegistration.props.isLoaded).toBeFalsy();

    // From here
    const renderVal = () => <View />;
    wrapper.setProps({
      children: (
        <UserRegistration
          data={{ movies: [{ title: 'star wars' }] }}
          onLoaded={jest.fn()}
          onLoading={jest.fn()}
          onConnecting={jest.fn()}
          navigation={{ navigate: jest.fn() }}
          render={renderVal}
        />
      )
    });

    userRegistration = wrapper
      .find('UserRegistration')
      .instance() as UserRegistration;

    userRegistration.componentDidUpdate();
    expect(userRegistration.props.data.movies[0].title).toBe('star wars');
    expect(userRegistration.props.navigation.navigate).toHaveBeenCalledTimes(1);
  });

  test('should set the error message on pressing start chat button', () => {
    // From here
    const renderVal = () => <View />;
    wrapper.setProps({
      children: (
        <UserRegistration
          data={{ errorMessage: 'Error Occured' }}
          onLoaded={jest.fn()}
          onLoading={jest.fn()}
          onConnecting={jest.fn()}
          navigation={{ navigate: jest.fn() }}
          render={renderVal}
        />
      )
    });

    userRegistration = wrapper
      .find('UserRegistration')
      .instance() as UserRegistration;

    userRegistration.componentDidUpdate();
    expect(userRegistration.props.data.errorMessage).toBe('Error Occured');
  });
});
