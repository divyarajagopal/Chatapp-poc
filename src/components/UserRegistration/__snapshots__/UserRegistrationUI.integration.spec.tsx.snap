// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRegistrationUI Component integration with child component tests should render the component as expected with store context 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UserRegistrationUI>
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Connect(UserRegistration)
          render={[Function]}
        >
          <UserRegistration
            data={Array []}
            isLoaded={false}
            onConnecting={[Function]}
            onLoaded={[Function]}
            onLoading={[Function]}
            render={[Function]}
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Component
                  style={
                    Object {
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <Component
                      animating={true}
                      color="#999999"
                      hidesWhenStopped={true}
                      size="large"
                    >
                      <ActivityIndicator
                        animating={true}
                        color="#999999"
                        hidesWhenStopped={true}
                        size="large"
                      />
                    </Component>
                  </View>
                </Component>
                <Component
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  >
                    <TextInputUI
                      error=""
                      onChangeText={[Function]}
                      placeholder="Username"
                      value=""
                    >
                      <TextInput
                        allowFontScaling={true}
                        error=""
                        onChangeText={[Function]}
                        placeholder="Username"
                        placeholderTextColor="grey"
                        style={
                          Array [
                            Object {
                              "borderColor": "grey",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "margin": 10,
                              "padding": 10,
                              "width": "100%",
                            },
                            null,
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      >
                        <TextInput
                          allowFontScaling={true}
                          error=""
                          onChangeText={[Function]}
                          placeholder="Username"
                          placeholderTextColor="grey"
                          style={
                            Array [
                              Object {
                                "borderColor": "grey",
                                "borderRadius": 5,
                                "borderWidth": 3,
                                "margin": 10,
                                "padding": 10,
                                "width": "100%",
                              },
                              null,
                            ]
                          }
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </TextInput>
                    </TextInputUI>
                    <TextInputUI
                      error=""
                      onChangeText={[Function]}
                      placeholder="Email"
                      value=""
                    >
                      <TextInput
                        allowFontScaling={true}
                        error=""
                        onChangeText={[Function]}
                        placeholder="Email"
                        placeholderTextColor="grey"
                        style={
                          Array [
                            Object {
                              "borderColor": "grey",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "margin": 10,
                              "padding": 10,
                              "width": "100%",
                            },
                            null,
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      >
                        <TextInput
                          allowFontScaling={true}
                          error=""
                          onChangeText={[Function]}
                          placeholder="Email"
                          placeholderTextColor="grey"
                          style={
                            Array [
                              Object {
                                "borderColor": "grey",
                                "borderRadius": 5,
                                "borderWidth": 3,
                                "margin": 10,
                                "padding": 10,
                                "width": "100%",
                              },
                              null,
                            ]
                          }
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </TextInput>
                    </TextInputUI>
                    <TouchableOpacityUI>
                      <TouchableOpacity
                        activeOpacity={0.2}
                        style={
                          Array [
                            Object {
                              "borderColor": "grey",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "margin": 10,
                              "width": "100%",
                            },
                            undefined,
                          ]
                        }
                      >
                        <AnimatedComponent
                          accessible={true}
                          isTVSelectable={true}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "borderColor": "grey",
                                  "borderRadius": 5,
                                  "borderWidth": 3,
                                  "margin": 10,
                                  "width": "100%",
                                },
                                undefined,
                              ],
                              Object {
                                "opacity": 1,
                              },
                            ]
                          }
                        >
                          <Component
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "borderColor": "grey",
                                "borderRadius": 5,
                                "borderWidth": 3,
                                "margin": 10,
                                "opacity": 1,
                                "width": "100%",
                              }
                            }
                          >
                            <View
                              accessible={true}
                              isTVSelectable={true}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "borderColor": "grey",
                                  "borderRadius": 5,
                                  "borderWidth": 3,
                                  "margin": 10,
                                  "opacity": 1,
                                  "width": "100%",
                                }
                              }
                            >
                              <Button
                                disabled={true}
                                onPress={[Function]}
                                title="Start Chat"
                              >
                                <TouchableOpacity
                                  accessibilityRole="button"
                                  accessibilityStates={
                                    Array [
                                      "disabled",
                                    ]
                                  }
                                  activeOpacity={0.2}
                                  disabled={true}
                                  onPress={[Function]}
                                >
                                  <AnimatedComponent
                                    accessibilityRole="button"
                                    accessibilityStates={
                                      Array [
                                        "disabled",
                                      ]
                                    }
                                    accessible={true}
                                    isTVSelectable={true}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Array [
                                        undefined,
                                        Object {
                                          "opacity": 1,
                                        },
                                      ]
                                    }
                                  >
                                    <Component
                                      accessibilityRole="button"
                                      accessibilityStates={
                                        Array [
                                          "disabled",
                                        ]
                                      }
                                      accessible={true}
                                      isTVSelectable={true}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "opacity": 1,
                                        }
                                      }
                                    >
                                      <View
                                        accessibilityRole="button"
                                        accessibilityStates={
                                          Array [
                                            "disabled",
                                          ]
                                        }
                                        accessible={true}
                                        isTVSelectable={true}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Object {
                                            "opacity": 1,
                                          }
                                        }
                                      >
                                        <Component
                                          style={
                                            Array [
                                              Object {},
                                              Object {},
                                            ]
                                          }
                                        >
                                          <View
                                            style={
                                              Array [
                                                Object {},
                                                Object {},
                                              ]
                                            }
                                          >
                                            <Component
                                              disabled={true}
                                              style={
                                                Array [
                                                  Object {
                                                    "color": "#007AFF",
                                                    "fontSize": 18,
                                                    "padding": 8,
                                                    "textAlign": "center",
                                                  },
                                                  Object {
                                                    "color": "#cdcdcd",
                                                  },
                                                ]
                                              }
                                            >
                                              <Text
                                                disabled={true}
                                                style={
                                                  Array [
                                                    Object {
                                                      "color": "#007AFF",
                                                      "fontSize": 18,
                                                      "padding": 8,
                                                      "textAlign": "center",
                                                    },
                                                    Object {
                                                      "color": "#cdcdcd",
                                                    },
                                                  ]
                                                }
                                              >
                                                Start Chat
                                              </Text>
                                            </Component>
                                          </View>
                                        </Component>
                                      </View>
                                    </Component>
                                  </AnimatedComponent>
                                </TouchableOpacity>
                              </Button>
                            </View>
                          </Component>
                        </AnimatedComponent>
                      </TouchableOpacity>
                    </TouchableOpacityUI>
                  </View>
                </Component>
              </View>
            </Component>
          </UserRegistration>
        </Connect(UserRegistration)>
      </View>
    </Component>
  </UserRegistrationUI>
</Provider>
`;
